src/pages/index.js

```diff
@@ -4,54 +4,148 @@ import Section from "../components/Section.js";
 import PopupWithImage from "../components/PopupWithImage.js";
 import UserInfo from "../components/UserInfo.js";
 import PopupWithForm from "../components/PopupWithForm.js";
+import PopupWithConfirm from "../components/PopupWithConfirm.js";
 import "./index.css";
-import { config, initialCards } from "../utils/constants.js";
+import { config } from "../utils/constants.js";
+import { api } from "../components/Api.js";
+import Popup from "../components/Popup.js";
 
 const cardList = document.querySelector(".cards__list");
 
 const pictureModal = new PopupWithImage(".modal-picture");
 pictureModal.setEventListeners();
-const cardSection = new Section(
-  {
-    items: initialCards,
-    renderer: (cardData) => {
-      cardSection.addItem(createCard(cardData));
-    },
-  },
-  ".cards__list"
-);
-cardSection.renderItems();
+// const cardSection = new Section(
+//   {
+//     items: initialCards,
+//     renderer: (cardData) => {
+//       cardSection.addItem(createCard(cardData));
+//     },
+//   },
+//   ".cards__list"
+// );
+// cardSection.renderItems();
+
+const deleteModal = new PopupWithConfirm(".modal-delete");
+deleteModal.setEventListeners();
 
 const userInfo = new UserInfo({
   userNameSelector: ".profile__title",
-  userJobSelector: ".profile__description",
+  userAboutSelector: ".profile__description",
+  userAvatarSelector: ".profile__image",
 });
+
+let userId;
+
+Promise.all([api.getUserInfo(), api.getInitialCards()])
+  .then((values) => {
+    userId = values[0]._id;
+    const userData = values[0];
+
+    const initialCards = values[1];
+
+    userInfo.setUserInfo({
+      name: userData.name,
+      about: userData.about,
+      avatar: userData.avatar,
+    });
+    const cardSection = new Section(
+      {
+        items: initialCards,
+        renderer: (cardData) => {
+          cardSection.addItem(createCard(cardData));
+        },
+      },
+      ".cards__list"
+    );
+    cardSection.renderItems();
+    // return userData;
+  })
+  .catch((err) => {
+    console.error("Error. The request has failed: ", err);
+  });
+
 const profileModal = new PopupWithForm(".modal-profile", (inputs) => {
-  userInfo.setUserInfo(inputs);
-  profileModal.close();
+  profileModal.changeButtonText("Saving...");
+  api
+    .updateUserInfo(inputs)
+    .then((res) => {
+      userInfo.setUserInfo({
+        name: res.name,
+        about: res.about,
+        avatar: res.avatar,
+      });
+      modalFormProfile.reset();
```

Comment:
`      modalFormProfile.reset();
`

Please, delete it

* You need to have method `close` inside `PopupWithForm` where you need to reset the form according to the task